com.vrcfury.vrcfury/Editor/VF/Feature/RemoveBlendshapeVerticesBuilder.cs

```diff
@@ -0,0 +1,171 @@
+using System.Collections.Generic;
+using System.Linq;
+using UnityEditor;
+using UnityEngine;
+using UnityEngine.UIElements;
+using VF.Builder;
+using VF.Feature.Base;
+using VF.Inspector;
+using VF.Model.Feature;
+
+namespace VF.Feature {
+
+public class RemoveBlendshapeVerticesBuilder : FeatureBuilder<RemoveBlendshapeVertices> {
+    [FeatureBuilderAction]
+    public void Apply() {
+        var renderers = new List<SkinnedMeshRenderer>();
+        if (model.allRenderers) {
+            renderers.AddRange(avatarObject.GetComponentsInSelfAndChildren<SkinnedMeshRenderer>());
+        } else {
+            renderers.Add(model.renderer);
+        }
+
+        foreach (var renderer in renderers) {
+            var original = renderer.sharedMesh;
+            if (original == null)
+                continue;
+            var blendshapeIndex = original.GetBlendShapeIndex(model.blendshape);
+            if (blendshapeIndex < 0)
+                continue;
+
+            var mesh = MutableManager.MakeMutable(original, forceCopy: true);
+            var vertexBuffer = new Vector3[mesh.vertexCount];
+            mesh.GetBlendShapeFrameVertices(blendshapeIndex, 0, vertexBuffer, null, null);
+            var toRemove = vertexBuffer.Select(v => !v.Equals(Vector3.zero)).ToArray();
+            try
+            {
+                RemoveVerticesFromMesh(toRemove, original, mesh);
+                renderer.sharedMesh = mesh;
+            }
+            catch (System.Exception ex)
+            {
+                Debug.LogException(ex);
+            }
+        }
+    }
+
+    // adapted from: https://github.com/euan142/LazyOptimiser/blob/main/Packages/net.euan.lazyoptimiser/Editor/MeshUtil.cs
+    private static void RemoveVerticesFromMesh(bool[] verticesToRemove, Mesh original, Mesh mesh) {
```

Comment:
Really not a fan of `original` getting passed in here. Preferably you'd just pull out everything you need from `mesh`, save it, and use it later when needed.